frontend/src/pages/collections/collection.ts

```diff
@@ -48,27 +48,36 @@ export class Collection extends BtrixElement {
   };
 
   private readonly orgCollections = new Task(this, {
-    task: async ([slug]) => {
-      if (!slug) throw new Error("slug required");
-      const org = await this.fetchCollections({ slug });
+    task: async ([orgSlug]) => {
+      if (!orgSlug) throw new Error("orgSlug required");
+      const org = await this.fetchCollections({ orgSlug });
       return org;
     },
-    args: () => [this.slug] as const,
+    args: () => [this.orgSlug] as const,
   });
 
   private readonly collection = new Task(this, {
-    task: async ([slug, collectionId]) => {
-      if (!slug || !collectionId)
-        throw new Error("slug and collection required");
-      const collection = await this.fetchCollection({ slug, collectionId });
+    task: async ([orgSlug, collectionSlug]) => {
+      if (!orgSlug || !collectionSlug)
+        throw new Error("orgSlug and collection required");
+      const collection = await this.fetchCollection({
+        orgSlug,
+        collectionSlug,
+      });
+
+      if (collection.redirectToSlug) {
```

Comment:
I think we don't actually need redirectToSlug at all, since can simply check this instead:

```suggestion
  if (collection.slug !== collectionSlug) {
```

Will simplify!

Reply:
Updated in latest commit!